tls/s2n_connection.c

```diff
@@ -834,23 +834,32 @@ int s2n_connection_use_corked_io(struct s2n_connection *conn)
     return 0;
 }
 
-uint64_t s2n_connection_get_wire_bytes_in(struct s2n_connection * conn)
+uint64_t s2n_connection_get_wire_bytes_in(struct s2n_connection *conn)
 {
+    notnull_check(conn);
```

Comment:
The `notnull_check` Macro returns `-1` if `conn` is null, but the function return type is an unsigned `uint64_t`. The return type needs to be signed for this Macro to work, so we probably need to change the return type of these functions to `int64_t` or `ssize_t`.

Reply:
We have a backlog issue to add the `-Wconversion` flag to our build that would have caught this issue and failed the build: https://github.com/awslabs/s2n/issues/789

Reply:
I cut out the three `notnull_check`'s added to `uinit64_t` returning functions for now. I can add them back in a separate PR if we want, but need to continue the discussion about api changes.